gtsfm/common/gtsfm_data.py

```diff
@@ -248,27 +248,32 @@ def from_selected_cameras(cls, gtsfm_data: "GtsfmData", camera_indices: List[int
 
         return new_data
 
-    def get_scene_avg_reprojection_error(self) -> float:
+    def get_avg_scene_reprojection_error(self) -> float:
         """Get average reprojection error for all 3d points in the entire scene
 
         Returns:
-            scene_avg_reproj_error: average of reprojection errors for every 3d point to its 2d measurements
+            Average of reprojection errors for every 3d point to its 2d measurements
         """
-        scene_reproj_errors = []
+        scene_reproj_errors: List[float] = []
         for track in self._tracks:
             track_errors, _ = reproj_utils.compute_track_reprojection_errors(self._cameras, track)
             scene_reproj_errors.extend(track_errors)
 
-        scene_avg_repoj_error = np.mean(scene_reproj_errors)
+        scene_avg_reproj_error = np.mean(scene_reproj_errors)
+
+        return scene_avg_reproj_error
 
+    def log_scene_reprojection_error_stats(self) -> None:
+        """Logs reprojection error stats for all 3d points in the entire scene."""
+        scene_reproj_errors = []
+        for track in self._tracks:
```

Comment:
maybe even cleaner is to have 3 functions:

```python
def get_track_reprojection_errors():
        for track in self._tracks:
            track_errors, _ = reproj_utils.compute_track_reprojection_errors(self._cameras, track)
            scene_reproj_errors.extend(track_errors)
      return scene_Reproj_errors
```

and the average getter and logger both call this.

No redundant code that way @ayushbaid